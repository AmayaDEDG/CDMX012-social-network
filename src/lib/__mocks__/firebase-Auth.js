/* eslint-disable max-classes-per-file */
global.firebase = {
  initializeApp: () => {
  },
  auth: () => ({
    signInWithEmailAndPassword: () => new Promise((resolve) => {
      resolve('');
    }),
    createUserWithEmailAndPassword: () => new Promise((resolve) => {
      resolve('');
    }),
    signInWithPopup: () => new Promise((resolve) => {
      resolve('');
    }),
    signInWithGoogle: () => {
      class GoogleAuthProvider {
        constructor() {
          this.id = 'email';
        }
      }
    },
    signInWithFacebook: () => {
      class FacebookAuthProvider {
        constructor() {
          this.id = 'email';
        }
      }
    },
  }),
  firestore: () => {
  },
};

export default firebase;
