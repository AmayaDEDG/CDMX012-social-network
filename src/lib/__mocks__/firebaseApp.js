global.firebase = {
  initializeApp: () => {
  },
  auth: () => {
  },
//   firestore: () => {
//   },
//   // signInEmailAndPW: () => {
//   //   const signInWithEmailAndPassword = () => {};
//   // },
//   signInWithGoogle: () => {
//     const GoogleAuthProvider = () => {
//     };
//   },
//   signInWithFacebook: () => {
//     const FacebookAuthProvider = () => {
//     };
//   },
};
