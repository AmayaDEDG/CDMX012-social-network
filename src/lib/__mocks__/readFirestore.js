global.firebase = {
  initializeApp: () => {
  },
  auth: () => ({
    onAuthStateChanged: () => {
    },
  }),
  accessPosts: () => {

  },
  firestore: () => {
  },
};

export default firebase;
