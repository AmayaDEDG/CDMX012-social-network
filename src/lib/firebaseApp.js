/* eslint-disable arrow-body-style */
/* eslint-disable no-underscore-dangle */
/* eslint-disable max-len */
/* eslint-disable import/no-cycle */
/* eslint-disable no-param-reassign */
/* eslint-disable no-unused-vars */
//  Firebase implementation
const firebaseConfig = {
  apiKey: 'AIzaSyC26n4Fh-NfxC_ZNKZrFDH4NzrQrYwgirY',
  authDomain: 'petspace-3f65f.firebaseapp.com',
  projectId: 'petspace-3f65f',
  storageBucket: 'petspace-3f65f.appspot.com',
  messagingSenderId: '719999017536',
  appId: '1:719999017536:web:4e72654f1a5dba66b1b5a5',
};
firebase.initializeApp(firebaseConfig);
// firebase.firestore().settings({ experimentalForceLongPolling: true, merge: true });

export const getCurrentUserName = () => {
  const user = firebase.auth().currentUser;
  const userName = user.displayName;
  return userName;
};
