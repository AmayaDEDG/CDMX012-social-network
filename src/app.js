/* eslint-disable import/no-cycle */
//  Routing
import { home } from './home.js';
import { registration } from './registration.js';

//  Firebase implementation
const firebaseConfig = {
  apiKey: 'AIzaSyC26n4Fh-NfxC_ZNKZrFDH4NzrQrYwgirY',
  authDomain: 'petspace-3f65f.firebaseapp.com',
  projectId: 'petspace-3f65f',
  storageBucket: 'petspace-3f65f.appspot.com',
  messagingSenderId: '719999017536',
  appId: '1:719999017536:web:4e72654f1a5dba66b1b5a5',
};
export const firebaseInitialization = firebase.initializeApp(firebaseConfig);

export const errorTranslate = {
  'auth/invalid-email': 'El email es inválido.',
  'auth/email-already-in-use': 'El email ya está registrado.',
  'auth/weak-password': 'La contraseña es inválida',
  'auth/wrong-password': 'La contraseña es incorrecta',
  'auth/user-not-found': 'El usuario no existe',
};

//  Routing
const routes = {
  '/': home,
  '/registration': registration,
};

const root = document.getElementById('root');
root.appendChild(routes[window.location.pathname]());

export const onNavigate = (pathname) => {
  window.history.pushState(
    {},
    pathname,
    window.location.origin + pathname,
  );
  if (root.hasChildNodes()) {
    root.innerHTML = '';
  }
  root.appendChild(routes[pathname]());
};

window.onpopstate = () => {
  root.appendChild(routes[window.location.pathname]());
};
