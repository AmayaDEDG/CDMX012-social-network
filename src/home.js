/* eslint-disable no-console */
/* eslint-disable no-trailing-spaces */
/* eslint-disable import/no-cycle */
import { onNavigate } from './app.js';
import { firebaseAuth/*, firebaseInitialization*/ } from './lib/firebaseAppJS.js';

export const home = () => {
  const petspaceLogoAttributes = {
    class: 'petspace-logo',
    id: 'petspaceLogo',
    src: 'img/PetSpaceLogo.png',
  };
  const loginBoxAttributes = {
    class: 'login-box',
    id: 'loginBox',
  };
  const loginFormAttributes = {
    class: 'login-form',
    id: 'loginForm',
  };
  const loginMailAttributes = {
    id: 'user',
    name: 'user',
    type: 'text',
    placeholder: '  Usuarix / e-mail',
  };
  const loginPWAttributes = {
    id: 'loginPassword',
    type: 'password',
    placeholder: '  Constraseña',
  };
  const messageAttributes = {
    id: 'loginMessage',
  };
  const seePWBtnAttributes = {
    id: 'seePassword',
  };
  const seePWBtnIAttributes = {
    class: 'fas fa-eye',
  };
  const loginSubmitAttributes = {
    class: 'submit-button',
    id: 'submitButton',
    type: 'submit',
  };
  const loginWithSectionAttributes = {
    class: 'login-with',
    id: 'loginWith',
  };
  const facebookLogoAttributes = {
    class: 'facebook-logo',
    id: 'facebookLogo',
    src: 'img/facebook.png',
  };
  const gmailLogoAttributes = {
    class: 'gmail-logo',
    id: 'gmailLogo',
    src: 'img/google.png',
  };
  const registerSectionAttributes = {
    class: 'register-section',
    id: 'registerSection',
  };
  const registerBtnAttributes = {
    class: 'register-button',
    id: 'registerButton',
    type: 'button',
  };

  const setAttributes = (element, attributes) => {
    Object.keys(attributes).forEach((attr) => element.setAttribute(attr, attributes[attr]));
  };

  const homeStructure = document.createElement('div');
  const petspaceLogo = document.createElement('img');
  const loginBox = document.createElement('section');
  const loginForm = document.createElement('form');
  const loginMail = document.createElement('input');
  const loginPW = document.createElement('input');
  const loginMessage = document.createElement('p');
  const seePWBtn = document.createElement('button');
  const seePassword = document.createElement('i');
  const loginSubmit = document.createElement('button');
  const loginWithSection = document.createElement('section');
  const loginWithText = document.createElement('p');
  const facebookLogo = document.createElement('img');
  const gmailLogo = document.createElement('img');
  const registerSection = document.createElement('section');
  const registerSectionText = document.createElement('p');
  const registerBtn = document.createElement('button');
  const addBr = document.createElement('br');

  setAttributes(petspaceLogo, petspaceLogoAttributes);
  setAttributes(loginBox, loginBoxAttributes);
  setAttributes(loginForm, loginFormAttributes);
  setAttributes(loginMail, loginMailAttributes);
  setAttributes(loginPW, loginPWAttributes);
  setAttributes(loginMessage, messageAttributes);
  setAttributes(seePWBtn, seePWBtnAttributes);
  setAttributes(seePassword, seePWBtnIAttributes);
  setAttributes(loginSubmit, loginSubmitAttributes);
  setAttributes(loginWithSection, loginWithSectionAttributes);
  setAttributes(facebookLogo, facebookLogoAttributes);
  setAttributes(gmailLogo, gmailLogoAttributes);
  setAttributes(registerSection, registerSectionAttributes);
  setAttributes(registerBtn, registerBtnAttributes);

  loginWithText.textContent = 'Inicia sesión con:';
  registerSectionText.textContent = '¿No tienes cuenta?';
  registerBtn.textContent = 'Regístrate';
  loginSubmit.textContent = 'Entrar';
  
  seePWBtn.appendChild(seePassword);
  loginForm.append(loginMail, loginPW, addBr, loginMessage, seePWBtn, loginSubmit);
  loginBox.appendChild(loginForm);
  loginWithSection.append(loginWithText, facebookLogo, gmailLogo);
  registerSection.append(registerSectionText, registerBtn);
  homeStructure.append(petspaceLogo, loginBox, loginWithSection, registerSection);

  seePWBtn.addEventListener('click', (e) => {
    e.preventDefault();
    if (loginPW.type === 'password') {
      loginPW.type = 'text';
    } else {
      loginPW.type = 'password';
    }
  });
  registerBtn.addEventListener('click', () => { onNavigate('/registration'); });
  // firebaseInitialization();
  loginSubmit.addEventListener('click', (e) => {
    e.preventDefault();
    const email = loginMail.value;
    const password = loginPW.value;
    const message = loginMessage;
    console.log(email, password);
    firebaseAuth(email, password, message);
  });
  return homeStructure;
};
