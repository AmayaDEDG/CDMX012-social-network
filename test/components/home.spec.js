/**
 * @jest-environment jsdom
 */
import '../../src/lib/__mocks__/firebaseApp';
import '../app';
import { home } from '../../src/components/home.js';

describe('home', () => {
  it('Snapshot home', () => {
    const component = home();
    document.body.innerHTML = '<div id="root"></div>';
    const rootDiv = document.getElementById('root');
    rootDiv.appendChild(component);
    expect(rootDiv.innerHTML).toMatchSnapshot();
  });
  it('has functionality in button to see Password', () => {
    expect()
  });
  it('Ã­s a function', () => {
    expect(typeof home).toBe('function');
  });
});
