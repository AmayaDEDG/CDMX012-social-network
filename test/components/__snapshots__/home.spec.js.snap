// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`home Snapshot home 1`] = `"<div class=\\"structure\\"><img class=\\"desktop-img\\" src=\\"img/homeDesktop.png\\"><div class=\\"home-structure\\"><img class=\\"petspace-logo\\" id=\\"petspaceLogo\\" src=\\"img/PetSpaceLogo.png\\"><section class=\\"login-box\\" id=\\"loginBox\\"><form class=\\"login-form\\" id=\\"loginForm\\"><input id=\\"user\\" name=\\"user\\" type=\\"text\\" placeholder=\\"E-mail\\"><input id=\\"loginPassword\\" type=\\"password\\" placeholder=\\"Contraseña\\"><p id=\\"loginMessage\\"></p><button id=\\"seePassword\\"><i class=\\"fas fa-eye\\"></i></button><button class=\\"submit-button\\" id=\\"submitButton\\" type=\\"submit\\">Entrar</button></form></section><section class=\\"login-with\\" id=\\"loginWith\\"><p>Inicia sesión con:</p><img class=\\"facebook-logo\\" id=\\"facebookLogo\\" src=\\"img/facebook.png\\"><img class=\\"gmail-logo\\" id=\\"gmailLogo\\" src=\\"img/google.png\\"></section><section class=\\"register-section\\" id=\\"registerSection\\"><p>¿No tienes cuenta?</p><button class=\\"register-button\\" id=\\"registerButton\\" type=\\"button\\">Regístrate</button></section></div></div>"`;
